dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.40/V2_40_13__Attribute_flags_as_jsonb.sql

```diff
@@ -0,0 +1,107 @@
+/* Converts numerous attribute boolean columns into a single jsonb column
+   with an array containing enum names of those columns that were true */
+
+/* create jsonb set column */
+alter table attribute add column if not exists objecttypes jsonb;
+
+/* make sure we have a function to create jsonb arrays without nulls */
+create or replace function jsonb_build_array_without_nulls(variadic anyarray)
```

Comment:
This is nice.

Reply:
There is surprisingly little support for filtering nulls in the standard posgres functions. You can filter object members with null values but not arrays. And you cannot convert object keys to an array within an update. Some simple obvious operations do not yet exist in the standard functions, which is a bit surprising.